src/Microsoft.DocAsCode.Build.RestApi/RestApiHelper.cs

```diff
@@ -45,41 +47,57 @@ public static SwaggerFormattedReference FormatReferenceFullPath(string reference
             // Decode for URI Fragment Identifier Representation
             if (reference.StartsWith("#/"))
             {
-                // Reuse relative path, to decode the values inside '/'.
-                var path = reference.Substring(2);
-                var decodedPath = ((RelativePath)path).UrlDecode();
-                return new SwaggerFormattedReference
+                var result = ParseReferencePath(reference.Substring(2));
+                return new SwaggeraInternalFormattedReference
                 {
-                    Type = SwaggerFormattedReferenceType.InternalReference,
-                    Path = "/" + decodedPath,
-                    Name = decodedPath.FileName
+                    Path = "/" + result.Item1,
+                    Name = result.Item2
                 };
             }
 
-            // External reference json
-            if (reference.EndsWith(".json", StringComparison.OrdinalIgnoreCase))
+            // Not decode for JSON String Representation
+            if (reference.StartsWith("/"))
             {
-                return new SwaggerFormattedReference
+                return new SwaggeraInternalFormattedReference
                 {
-                    Type = SwaggerFormattedReferenceType.ExternalReference,
                     Path = reference,
-                    Name = Path.GetFileNameWithoutExtension(reference)
+                    Name = reference.Split('/').Last()
                 };
             }
 
-            // Not decode for JSON String Representation
-            if (reference.StartsWith("/"))
+            // External reference
+            if (PathUtility.IsRelativePath(reference) && reference.Contains(JsonExtension))
             {
-                var fileName = reference.Split('/').Last();
-                return new SwaggerFormattedReference
+                // For example "file.json"
+                if (reference.EndsWith(JsonExtension))
```

Comment:
just realize the json pointer should be case sensitive.

---
In reply to: [101434772](https://github.com/dotnet/docfx/pull/1308#discussion_r101434772) [](ancestors = 101434772)